File: src/junghyun/discord/ui/languages/LanguageInterface.java
package junghyun.discord.ui.languages;

public interface LanguageInterface {

    String[] TARGET_REGION();

    String LANGUAGE_CODE();

    String LANGUAGE_NAME();
    String LANGUAGE_DESCRIPTION();

    String HELP_INFO();
    String HELP_DESCRIPTION();
    String HELP_DEV();
    String HELP_GIT();
    String HELP_VERSION();
    String HELP_SUPPORT();
    String HELP_INVITE_LINK();

    String HELP_CMD_INFO();
    String HELP_CMD_HELP();
    String HELP_CMD_LANG(String langList);
    String HELP_CMD_SKIN();
    String HELP_CMD_RANK();
    String HELP_CMD_PVE();
    String HELP_CMD_PVP();
    String HELP_CMD_RESIGN();

    String SKIN_INFO();
    String SKIN_DESCRIPTION();
    String SKIN_CMD_INFO(String style);
    String SKIN_CHANGE_ERROR();
    String SKIN_CHANGE_SUCCESS(String style);

    String RANK_INFO();
    String RANK_DESCRIPTION();
    String RANK_WIN();
    String RANK_LOSE();

    String LANG_CHANGE_ERROR();
    String LANG_SUCCESS();

    String GAME_NOT_FOUND(String nameTag);
    String GAME_CREATE_FAIL(String nameTag);
    String GAME_SYNTAX_FAIL(String nameTag);
    String GAME_ALREADY_IN(String nameTag);

    String GAME_CREATE_INFO(String playerName, String targetName, String fAttack);
    String GAME_CMD_INFO();

    String GAME_NEXT_TURN(String curName, String prvName, String lastPos);

    String GAME_PVP_TURN(String turnName);
    String GAME_PVP_WIN(String winName, String loseName, String lastPos);
    String GAME_PVP_RESIGN(String winName, String loseName);
    String GAME_PVP_INFO(String winName, String loseName, int winCount, int loseCount);

    String GAME_PVE_WIN(String lastPos);
    String GAME_PVE_LOSE(String lastPos);
    String GAME_PVE_RESIGN();
    String GAME_PVE_INFO(String playerName, int winCount, int loseCount, int rank);

    String GAME_FULL();

    String GAME_ARCHIVED(String messageLink);

    String BOARD_INP();
    String BOARD_FINISH();
    String BOARD_TURNS();
    String BOARD_TURN();
    String BOARD_LOCATION();

}

File: src/main/kotlin/i18n/LanguageInterface.kt
package i18n

interface LanguageInterface {

    fun targetRegion(): Array<String>

    fun languageCode(): String

    fun languageName(): String
    fun languageDescription(): String

    fun helpInfo(): String
    fun helpDescription(): String
    fun helpDev(): String
    fun helpRepo(): String
    fun helpVersion(): String
    fun helpSupport(): String
    fun helpInvite(): String

    fun helpCmdInfo(): String
    fun helpCmdHelp(): String
    fun helpCmdLang(langList: String): String
    fun helpCmdSkin(): String
    fun helpCmdRank(): String
    fun helpCmdPVE(): String
    fun helpCmdPVP(): String
    fun helpCmdResign(): String

    fun skinInfo(): String
    fun skinDescription(): String
    fun skinCmdInfo(style: String): String
    fun skinUpdateError(): String
    fun skinUpdateSuccess(style: String): String

    fun rankInfo(): String
    fun rankDescription(): String
    fun rankWin(): String
    fun rankLose(): String

    fun langUpdateError(): String
    fun langUpdateSuccess(): String

    fun gameNotFound(nameTag: String): String
    fun gameCreationFail(nameTag: String): String
    fun gameSyntaxFail(nameTag: String): String
    fun gameAlreadyInProgress(nameTag: String): String

    fun gameCreateInfo(playerName: String, targetName: String, fAttack: String): String
    fun gameCmdInfo(): String

    fun gameNextTurn(curName: String, prvName: String, lastPos: String): String

    fun gamePVPNextTurn(turnName: String): String
    fun gamePVPWin(winName: String, loseName: String, lastPos: String): String
    fun gamePVPResign(winName: String, loseName: String): String
    fun gamePVPInfo(winName: String, loseName: String, winCount: Int, loseCount: Int): String

    fun gamePVEWin(lastPos: String): String
    fun gamePVELose(lastPos: String): String
    fun gamePVEResign(): String
    fun gamePVEInfo(playerName: String, winCount: Int, loseCount: Int, rank: Int): String

    fun gameFull(): String

    fun boardInProgress(): String
    fun boardFinish(): String
    fun boardProgress(): String
    fun boardTurns(): String
    fun boardLocation(): String

}